framework/ChiPhysics/PhysicsMaterial/transportxsections/prop_transportxs_01b_finalize.cc

```diff
@@ -68,131 +80,192 @@ void chi_physics::TransportCrossSections::Finalize()
+        if (std::all_of(tmp[j].begin(), tmp[j].end(),
+                        [](double x) { return x == 0.0; }))
           throw std::logic_error(
-              "The delayed neutron precursor decay constants "
-              "was not provided.");
+              "Invalid delayed emission spectra encountered for "
+              "precursor species " + std::to_string(j) + ".\n" +
+              "Spectra must have at least one nonzero value.");
 
-        //throw error if not specified
-        if (!was_specified(precursor_yield))
-          throw std::logic_error(
-              "The delayed neutron precursor yields was not provided.");
+        //normalize each emission spectrum
+        double chi_delayed_j_sum =
+            std::accumulate(tmp[j].begin(), tmp[j].end(), 0.0);
+
+        for (unsigned int g = 0; g < num_groups; ++g)
+          chi_delayed[g][j] /= chi_delayed_j_sum;
+      }//for j
+
+      //==================================================
+      // Check the precursor data
+      //==================================================
+
+      if (precursor_lambda.empty())
+        throw std::logic_error("Precursor decay constants not found.");
+      if (!std::all_of(precursor_lambda.begin(),
+                       precursor_lambda.end(),
+                      [](double x) { return x > 0.0;}))
+        throw std::logic_error(
+            "Invalid precursor decay constant encountered.\n"
+            "Decay constants must be greater than zero.");
+
+      if (precursor_yield.empty())
+        throw std::logic_error("Precursor yield fractions no found.");
+      if (!std::all_of(precursor_yield.begin(),
+                       precursor_yield.end(),
+                       [](double x) { return x >= 0.0 && x <= 1.0; }))
+        throw std::logic_error(
+            "Invalid delayed neutron precursor yield fraction "
+            "encountered.\n"
+            "Yield fractions must be in the range [0.0, 1.0]");
+
+      //normalize the precursor yields
+      double yield_sum = std::accumulate(precursor_yield.begin(),
+                                         precursor_yield.end(), 0.0);
+
+      for (unsigned int j = 0; j < num_precursors; ++j)
+        precursor_yield[j] /= yield_sum;
 
-        //normalize the precursor yields
-        double yield_sum = std::accumulate(precursor_yield.begin(),
-                                           precursor_yield.end(), 0.0);
-        for (unsigned int j = 0; j < num_precursors; ++j)
-          precursor_yield[j] /= yield_sum;
-      }
 
-      //compute beta-weighted total fission spectrum
+      //==================================================
+      // Compute the steady-state fission spectrum
+      //==================================================
+
+      // This is computed via a beta-weighting
+      // NOTE: Generally, the computation of this quantity requires
+      //       a weight function. This estimate uses a unit weight
```

Comment:
isn't it the case that if nu_p and nu_dj are independent of energy, then this is actually exact?

Reply:
I believe that is true because then the the sum over the fission rate can be eliminated. I'll add this to the note.

Reply:
There were lots of instances of "cross-sections" in this directory! Most of which were not mine!